tests/mockers.py

```diff
@@ -95,9 +95,10 @@ def commit_amend(self, message: str) -> "GitRepositorySandbox":
         self.execute(f'git commit --amend -m "{message}"')
         return self
 
-    def push(self, remote: str = 'origin', set_upstream: bool = True) -> "GitRepositorySandbox":
+    def push(self, remote: str = 'origin', set_upstream: bool = True, tracking_branch: Optional[str] = None) -> "GitRepositorySandbox":
         branch = popen("git symbolic-ref -q --short HEAD")
-        self.execute(f"git push {'--set-upstream' if set_upstream else ''} {remote} {branch}")
+        tracking_branch = popen("git symbolic-ref -q --short HEAD") if not tracking_branch else tracking_branch
```

Comment:
Huh that's the same `popen` as in the line above 🤔

Reply:
Yeah, nice catch, will fix